deepspeed/inference/engine.py

```diff
@@ -122,7 +124,31 @@ def __init__(self, model, config):
                 self._apply_injection_policy(config, client_module)
         elif config.replace_method == 'auto':
             self._apply_injection_policy(config)
-
+        elif config.replace_method == 'dict':
+            key = ParserPolicies.get_map_key(str(model))
+            new_key = None
+            def SML(module):
+                for child in module.children():
+                    if isinstance(child, nn.ModuleList):
+                        return child[0].__class__
+                    else:
+                        new_key = SML(child)
+                    if new_key is not None:
+                        return new_key
+
+            new_key = SML(model)
+            if new_key is not None:
+                key = str(new_key).split('models.')[1].split('.')[0]
+            assert key in ParserPolicies.parser_policy_map, "dict replace method not supported for this model"
+            for client_module, injection_policy in ParserPolicies.parser_policy_map[key].items():
+                client_module = import_string('transformers.models.' + key +
```

Comment:
I’m not familiar with the ‘import_string’ package, but would could importlib be used here? I am always wary of adding new dependencies :)

https://docs.python.org/3.9/library/importlib.html#importlib.import_module

Reply:
I don't think import_module will work here because it can only import modules and I need a class object returned here which is not included in any module.

Reply:
Can we do what they’re recommending here: 

https://github.com/rochacbruno-archive/import_string

I see the package is deprecated as well, so if we can avoid using it that would probably be ideal

Reply:
Awesome, saw the updates, thanks @molly-smith!